### PR TITLE
Authentication against Central EGA's user database

### PR DESCRIPTION
**Pull request by silverdaz**
Updating the C code to connect to Central EGA and not its fake instance.

We use a user/password combination for the moment, to make sure not everyone can query CentralEGA (over https).

The response is JSON formatted. We parse the response and query parts of it using `libjq`. Queries are parametrized in the `auth.conf` file. So is the REST endpoint and the user/password combination.

### CODE REVIEW

**Comment by silverdaz** [comment]
Connected to #155

**Comment by silverdaz** [comment]
I forgot to mention here: Central EGA returns password hash using blowfish. We have a fallback to sha256, sha512 (and even md5) using the glibc crypt function.
Too bad blowfish is not include in GNU/Linux.
(actually: some distributions have it, like SuSE, but not all. CentOS doesn't have it.)

File: src/auth/cega.c [hunk 1]
```diff
@@ -69,50 +100,62 @@ fetch_from_cega(const char *username, char **buffer, size_t *buflen, int *errnop
...
   /* curl_easy_setopt(curl, CURLOPT_SSLCERTTYPE  , "PEM"            ); */
 
-#ifdef DEBUG
-  curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0L);
-  curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, 0L);
-#endif
+/* #ifdef DEBUG */
+/*   curl_easy_setopt(curl, CURLOPT_SSL_VERIFYPEER, 0L); */
+/*   curl_easy_setopt(curl, CURLOPT_SSL_VERIFYHOST, 0L); */
+/* #endif */
```
Comment (viklund) [inline src/auth/cega.c]: Remove if everything is commented out.

File: docker/ega.yml [hunk 2]
```diff
@@ -140,15 +142,15 @@ services:
...
-       - ${CEGA_USERS}:/cega/users:rw
+  # # Faking Central EGA
+  # cega_users:
+  #   build: images/cega_users
+  #   image: nbis/ega:cega_users
+  #   container_name: cega_users
+  #   ports:
+  #     - "9100:80"
+  #   volumes:
+  #      - ${CEGA_USERS}:/cega/users:rw
```
Comment (viklund) [inline docker/ega.yml]: Remove

File: docker/ega.yml [hunk 3]
```diff
@@ -44,8 +44,10 @@ services:
     hostname: ega_inbox
     depends_on:
       - db
-      - cega_users
-    env_file: .env.d/db
+#      - cega_users
```
Comment (viklund) [inline docker/ega.yml]: delete

**Pull request merged**